packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingUtils.ts

```diff
@@ -116,7 +119,7 @@ export const filterRowTreeFromGroupingColumns = (
           childNode,
 
           areAncestorsExpanded && !!node.childrenExpanded,
-          [...ancestorsResults, filterResults],
+          [...ancestorsResults, { ...filterResults }],
```

Comment:
Could we use the same reference as we likely won't be mutating it after this point. 🤔